pkg/apis/v1alpha1/wait.go

```diff
@@ -0,0 +1,65 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// OutputFormatType defines the enum for valid output formats.
+type OutputFormatType string
+
+const (
+	OutputFormatJSON           OutputFormatType = "json"
+	OutputFormatYAML           OutputFormatType = "yaml"
+	OutputFormatName           OutputFormatType = "name"
+	OutputFormatGoTemplate     OutputFormatType = "go-template"
+	OutputFormatGoTemplateFile OutputFormatType = "go-template-file"
+	OutputFormatTemplate       OutputFormatType = "template"
+	OutputFormatTemplateFile   OutputFormatType = "templatefile"
+	OutputFormatJSONPath       OutputFormatType = "jsonpath"
+	OutputFormatJSONPathAsJSON OutputFormatType = "jsonpath-as-json"
+	OutputFormatJSONPathFile   OutputFormatType = "jsonpath-file"
+)
+
+// For specifies the condition to wait for.
+type For struct {
+	// Deletion specifies parameters for waiting on a resource's deletion.
+	// +optional
+	Deletion *Deletion `json:"delete,omitempty"`
+	// Condition specifies the condition to wait for.
+	// +optional
+	Condition *Condition `json:"condition,omitempty"`
+}
+
+// Deletion represents parameters for waiting on a resource's deletion.
+type Deletion struct{}
+
+// Condition represents parameters for waiting on a specific condition of a resource.
+type Condition struct {
+	// ConditioName is the specific condition to wait for, e.g., "Available", "Ready".
+	ConditionName string `json:"conditionName"`
+	// The specific condition status to wait for, e.g., "True", "False".
+	// +optional
+	ConditionValue *bool `json:"conditionValue,omitempty"`
```

Comment:
```suggestion
	ConditionValue string `json:"conditionValue,omitempty"`
```

Reply:
why not bool ?

Reply:
Because `kubectl wait ...` expects a string

Reply:
`k wait pods --for=condition=Ready="something" -n local-path-storage --all`

Reply:
i manage it from here  

```
args = append(args, fmt.Sprintf("--for=condition=%s=%t", collector.For.Condition.ConditionName, *collector.For.Condition.ConditionValue))
```

Reply:
yeah but with a bool `Not yet` is not going to be accepted